api/campusdish_interface.py

```diff
@@ -14,33 +15,47 @@ def get_menu_data(location, meal_id, date):
     response = requests.get(
         f'https://uci.campusdish.com/api/menu/GetMenus?locationId={location_id}&periodId={MEAL_TO_PERIOD[meal_id][0]}&date={date}'
     )
-    if response.status_code==200:
+    if response.status_code == 200:
         payload = response.json()
         if 'Menu' in payload:
             return payload['Menu']
         else:
-            raise KeyError(f'Key "Menu" not found in campusdish response object. Response payload below:\n{payload}')
+            raise KeyError(
+                f'Key "Menu" not found in campusdish response object. Response payload below:\n{payload}')
     else:
         print("Response error message: ", response.json())
         response.raise_for_status()
 
-def get_schedule_data(location, date):
+
+def get_schedule_data(restaurant: str) -> dict:
     '''
-    Given a valid location and date,
-    perform get request for the schedule_json
+    Given the restaurant name,
+    perform get request, then parse the HTML code using BeautifulSoup 4
+    return a dictionary
+    schedule time use int because frontend work with int
+    schedule time is (100*hours)+minutes, where hours is in 24-hour time
     '''
-    response = requests.get(
-        f'https://uci.campusdish.com/api/menu/GetMenuPeriods?locationId={location}&date={date}'
-    )
-    if response.status_code==200:
-        return response.json()['Result']
+    url = 'https://uci.campusdish.com/LocationsAndMenus/'
+    if restaurant == 'Anteatery':
+        url += 'TheAnteatery'
     else:
-        print("Response error message: ", response.json())
-        response.raise_for_status()
+        url += restaurant
+    schedule = {}
+    soup = bs(requests.get(url).text, 'html.parser')
+    meal_period = soup.select('.mealPeriod')[:5]
+    location_times = soup.select('span[class=location__times]')[:5]
+    for idx, meal in enumerate(meal_period):
+        meal = meal.getText().lower()
+        times = location_times[idx].getText().split(' - ')
+        start = normalize_time_from_str(times[0])
```

Comment:
![image](https://user-images.githubusercontent.com/48658337/209749268-22d29f4d-4831-409e-8ce8-272405d170be.png)
Right now we're getting a crashing error when the helper function fails to parse the time. Currently it's because the dining halls are closed but seeing as this could also happen if they decide to change the website, we should make it fail without crashing when there's an error here.

Reply:
I didn't see the crashing error on my end. I think it is because the website has changed. Do you want me to add a big try and except around the function to prevent crashing and also print out the error?

Reply:
<img width="600" alt="image" src="https://user-images.githubusercontent.com/96160110/215422483-0aac6a50-b030-4b3a-9435-9675c9c33e4d.png">

I have another idea. The ValueError is caused by the invalid input time in the normalize_time_from_str(time), and the time is came from the times list. Do you think it will be a good idea if I turn the list into string and use regular expression to check if the times list is in the expected format? And if the time is in the expected format we can just do it as normal and return the schedule. If not, we will return the sample schedule. Do you think this will be good?

Reply:
Yeah, do that.